data/debug-cli/docker_debug.yaml

```diff
@@ -0,0 +1,322 @@
+command: docker debug
+short: Get a shell into any container or image. Replacement for `docker exec`.
+long: |-
+  > **Note**
+  >
+  > Docker Debug requires a [Pro, Team, or Business subcription](/subscription/details/).
+  > You must [sign in](/desktop/get-started/) to use this command.
+
+  Docker Debug helps you follow best practices by keeping your images small and secure. 
+  With Docker Debug, you can debug your images while they contain the bare minimum to run your application.
+  It does this by letting you create and work with slim images or containers that are often difficult to debug because all tools have been removed. 
+  For example, while typical debug approaches like `docker exec -it my-app bash` may not work on a slim container, `docker debug` will work.
+  
+  With `docker debug` you can get a debug shell into any container or image, even if they don't contain a shell. 
+  You don't need to modify the image to use Docker Debug. 
+  However, using Docker Debug still won't modify your image.
+  Docker Debug brings its own toolbox that you can easily customize.
+  The toolbox comes with many standard Linux tools pre-installed, such as `vim`, `nano`, `htop`, and `curl`.
+  Use the builtin `install` command to add additional tools available on https://search.nixos.org/packages.
+  Docker Debug supports `bash`, `fish`, and `zsh`. 
+  By default it tries to auto-detect your shell.
+  
+  
+  Custom builtin tools:
+  - `install [tool1] [tool2]`: Add Nix packages from: https://search.nixos.org/packages, see [example](#managing-your-toolbox-using-the-install-command).
+  - `uninstall [tool1] [tool2]`: Uninstall Nix packages.
+  - `entrypoint`: Print, lint, or run the entrypoint, see [example](#understanding-the-default-startup-command-of-a-container-entry-points).
+  - `builtins`: Show custom builtin tools.
+  
+  > **Note**
+  >
+  > For images and stopped containers, all changes are discarded when leaving the shell. 
+  > At no point, do changes affect the actual image or container. 
+  > When accessing running or paused containers, all filesystem changes are directly visible to the container. 
+  > The `/nix` directory is never visible to the actual image or container.
+
+
+usage: debug [OPTIONS] {CONTAINER|IMAGE}
```

Comment:
Note to self: we haven't used curly braces for denoting options in a required positional argument before. Afaik we've just used a pipe separator, unparenthesized.

https://github.com/docker/docs/blob/0791ea5e56953637965dfab0bd7a475685219f97/data/buildx/docker_buildx_build.yaml#L11

https://github.com/docker/docs/blob/0791ea5e56953637965dfab0bd7a475685219f97/data/engine-cli/docker_inspect.yaml#L60

Maybe these curly braces is a pattern we could adopt.
OTOH, many CLIs use angle brackets for denoting required commands (e.g. `jq`, `git`, `curl`), so there's an alternate presedence there.

```console
$ curl --help
Usage: curl [options...] <url>
```

From the before-mentioned examples, I think only Git allows different "types" to be used as the argument. In which case they've developed a higher-level concept that encapsulates all the options (e.g. [`commit-ish`](https://git-scm.com/docs/gitglossary#def_commit-ish), [`tree-ish`](https://git-scm.com/docs/gitglossary#def_tree-ish)).

Reply:
don't remember where the curly braces were coming from. At some point I had a discussion with someone at Docker who suggested me to use them to make it more clear that it's either or but required. And at that point it sounded plausible to me